### PR TITLE
Update AWS/Route53 reference setup terraform code

### PR DESCRIPTION
**Pull request by ytsarev**
* Update with the recent enhancements and terraform >= 1.0 compatibility
  fixes
* Documentation update

Signed-off-by: Yury Tsarev <yury.tsarev@absa.africa>

### CODE REVIEW

File: docs/deploy_route53.md [hunk 1]
```diff
@@ -8,8 +8,12 @@ Two EKS clusters in `eu-west-1` and `us-east-1`.
 
 Terraform code for cluster reference setup can be found [here](https://github.com/k8gb-io/k8gb/tree/master/docs/examples/route53)
 
-Feel free to reuse this code fully or partially and adapt for your existing scenario
-things like IRSA(IAM Roles for Service Accounts).
+Feel free to reuse this code fully or partially and adapt for your existing scenario custom configuration  like
```
Comment (jkremser) [inline docs/deploy_route53.md:11]: nit: `..iguration  like..` (2 spaces) Markdown + jekyll fixes this, but still

Comment (ytsarev) [reply docs/deploy_route53.md]: @jkremser amended, thanks for the catch.

File: docs/examples/route53/.gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
-.terraform
+.terraform*
+terraform.tfstate*
```
Comment (kuritka) [inline docs/examples/route53/.gitignore:2]: inspired by [Terraform.gitignore](https://github.com/github/gitignore/blob/master/Terraform.gitignore);  consider sensitive `*.tfvars`

Comment (ytsarev) [reply docs/examples/route53/.gitignore:2]: @kuritka we are not passing tfvars here at all, but doing it through the module invocation see `main.tf`

Comment (kuritka) [reply docs/examples/route53/.gitignore:2]: @ytsarev , ah, clear now. Thx

**Review by kuritka — DISMISSED** [review]

**Review by kuritka — APPROVED** [review]
lgtm

**Pull request merged**